### PR TITLE
Add `-o` option to unzip calls

### LINKED ISSUE #327: Workflow silently hangs if `basins_database` tries to unzip when `build/data/basins` is already populated
### What happened?

If you re-run the workflow after updates to the code, after having previously run it, leading to the situation where `build/data/basins` is already populated but snakemake wants to run the `basins_database` rule again, the workflow silently hangs (appearing as if the current rule is in progress, but in fact, indefinitely hanging).

On hitting enter, this output appears:

```
error:  invalid response [{ENTER}]
replace ./build/data/basins/hybas_eu_lev07_v1c.dbf? [y]es, [n]o, [A]ll, [N]one, [r]ename:
```

Either hitting `y` repeatedly or hitting `A` once lets unzip do it's job, and the workflow resumes.

Adding `-o` (silently overwrite) to unzip would solve this.

### Version

1.2.0.dev

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by brynpickering**
Fixes #327 

## Checklist

Any checks which are not relevant to the PR can be pre-checked by the PR creator. All others should be checked by the reviewer. You can add extra checklist items here if required by the PR.

- [x] CHANGELOG updated
- [ ] Minimal workflow tests pass
- [ ] Tests added to cover contribution
- [ ] Documentation updated
- [ ] Configuration schema updated

### CODE REVIEW

**Review by timtroendle — CHANGES_REQUESTED** [review]
I see how this is solving the issue #327. I would still be reluctant to merge this, because #327 _should_ never happen. Snakemake _should_ take care that the file to be generated is deleted before executing the rule.

I recommend to first better understand #327. Can we reproduce it? Maybe it's a bug in Snakemake? Maybe that bug is solved already in some newer version? Maybe we can solve the bug ourselves?

**Comment by brynpickering** [comment]
It happens for me too. It's because the `output` is just one of the files that is unzipped, not all of them. I can fix it properly in this instance, but there are other instances in which `-o` is used. 

EDIT: I have now cleaned these up to all be able to unzip without needing `-o` by only extracting the files that are actually needed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -70,6 +70,7 @@
 * **UPDATE** dropped support for Intel macOS. The workflow may still run on Intel macOS, but we do not actively maintain support (#369).
 * **UPDATE** link to GADM data following changes upstream (#376).
 * **UPDATE** link to NUTS data to allow for different NUTS years (#380, #382).
+* **UPDATE** unzip command to always overwrite existing data if it is called (#327).
```
Comment (timtroendle) [inline CHANGELOG.md:73]: This comment seems incorrect now. Maybe something like `FIX ensure only relevant data is unzipped (#)`?

**Review by timtroendle — APPROVED** [review]

File: rules/wind-and-solar.smk [hunk 2]
```diff
@@ -27,7 +27,7 @@ rule potentials:
         population = "build/data/{resolution}/population.csv",
         land_cover = "build/data/{resolution}/land-cover.csv"
     conda: "../envs/shell.yaml"
-    shell: "unzip -o {input} -d build/data"
+    shell: "for file in {output}; do unzip {input} ${{file#'build/data/'}} -d build/data; done"
```
[the remark this answers is not in the document]
Comment (brynpickering) [reply rules/wind-and-solar.smk:30]: oh nice, hadn't even clocked the `shadow` on this rule.

**Pull request merged**